_source/_posts/2022-04-21-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,669 @@
+
+If you already have an Okta account, use `okta login` to log into that account with the CLI. Otherwise, use `okta register` to sign up for a free account. 
+
+Create the OIDC app using the following command using a Bash shell opened to the project root.
+
+```bash
+okta apps create jhipster
+```
+
+You can accept the default values by pressing **enter**. If you want to give the app a different, more descriptive, name, that is fine. Just take note of the name because you will need to find the app in the Okta developer dashboard and update the redirect URIs a little later.
+
+This command created a config file named `.okta.env`. It will look something like the following. It helpfully lists the values you will need in the next step.
+
+```bas
+export SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_OIDC_ISSUER_URI="https://dev-123456/oauth2/default"
+export SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_ID="2989u928u383..."
+export SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_SECRET="09328uu098u4..."
+```
+
+Both the Docker Compose project and the Kubernetes project use Spring Cloud Config to centralize configuration. The JHipster registry service makes these config values available to all of the other services in the cluster.
+
+Open `docker-compose/central-server-config/application.yml` and add the following to the end, filling in the `issuer-uri`, `client-id`, and `client-secret` taken from the `.okta.env` file. This is the Spring Cloud Config file for the Docker Compose project.
+
+`docker-compose/central-server-config/application.yml`
+
+```yml
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          oidc:
+            issuer-uri: https://<your-okta-domain>/oauth2/default
+        registration:
+          oidc:
+            client-id: <client-id>
+            client-secret: <client-secret>
+```
+
+#### Build the Docker Images and Run the App with Docker Compose
+
+You're all set to run the app locally using Docker and Docker Compose. You need to build the docker image for each of the projects: `gateway`, `store`, and `blog`.
+
+ In the three different app directories, run the following Gradle command.
+
+```
+./gradlew -Pprod bootJar jibDockerBuild
+```
+
+Navigate to the `docker-compose` directory and run the app. You can use the `-d` param to run it as a daemon but for the moment I like seeing the logs. You're just running this in Docker Compose as a warm-up for the Azure deployment anyway.
+
+```bash
+docker-compose up
+```
+
+Give that a minute or two to finish running all the services. 
+
+Open the gateway service: http://localhost:8080/
+
+Go to **Account** and **Sign in**. You should be directed to the Okta sign-in form. 
+
+{% img blog/jhipster-k8s-azure/signing-in.png alt:"Okta Login" width:"500" %}{: .center-image }
+
+
+You should be able to authenticate with your Okta credentials.
+
+{% img blog/jhipster-k8s-azure/authenticated.png alt:"Authenticated" width:"800" %}{: .center-image }
```

Comment:
We might want to crop this on the right since there's a line on the right but not on the left.